libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/PushNotificationManager.swift

```diff
@@ -84,3 +89,100 @@ extension PushNotificationManager {
+        }
+    
+        private func createNotificationCategory(from type: NotificationType) -> UNNotificationCategory {
+            let actions = createActions(from: type.actionGroups)
+            return UNNotificationCategory(identifier: type.apiName, actions: actions, intentIdentifiers: [])
+        }
+    
+        private func createActions(from actionGroups: [ActionGroup]) -> [UNNotificationAction] {
+            return actionGroups.flatMap { actionGroup in
+                actionGroup.actions.map { action in
+                    UNNotificationAction(
+                        identifier: action.identifier,
+                        title: action.label,
+                        options: [.foreground] // Ensures the app opens if needed
+                    )
+                }
+            }
+        }
+}
+
+@objc
+extension PushNotificationManager {
+    /// Retrieves available notification types from the API or local cache if the API request fails.
+    ///
+    /// - Parameters:
+    ///   - restClient: The `RestClient` instance used to fetch notification types from the API.
+    ///   - userDefaults: Stores and retrieves cached notification types. Defaults to `.standard`.
+    /// - Returns: A tuple containing:
+    ///   - `Bool`: `true` if notification types were successfully retrieved and stored; `false` if retrieval failed.
+    ///   - `Error?`: An error object if retrieval fails completely, or `nil` if successful.
+    ///
+    /// If the API fetch fails, the function attempts to retrieve notification types from the cache.
+    /// If neither source provides data, the function returns `false` with an appropriate error.
+    @objc
+    public func getNotificationTypes(restClient: RestClient = RestClient.shared,
+                                     userDefaults: UserDefaults = .standard) async -> (Bool, Error?) {
+        do {
+            let types = try await fetchNotificationTypesFromAPI(with: restClient)
+            storeNotification(types: types, with: userDefaults)
+            setNotificationCategories(types: types)
+            return (true, nil)
+        } catch {
+            print("API fetch failed: \(error.localizedDescription). Trying cache...")
+            guard let cachedTypes = getCachedNotificationTypes(with: userDefaults) else {
+                print("No cached notification types available.")
+                return (false, PushNotificationManagerError.failedNotificationTypesRetrieval as Error)
+            }
+            print("Using cached notification types.")
+            setNotificationCategories(types: cachedTypes)
+            return (true, nil)
+        }
+    }
+    
+    /// Deletes all cached notification types and removes registered notification categories.
+    ///
+    /// - Parameter userDefaults: The `UserDefaults` instance used to remove cached notification types. Defaults to `.standard`.
+    @objc
+    public func deleteNotificationTypes(userDefaults: UserDefaults = .standard) {
```

Comment:
This to support the Logout

Reply:
See my previous comment. If you store it with the user account then the existing logout cleanup code will take care of it.